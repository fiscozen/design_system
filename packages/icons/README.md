# @fiscozen/icons

## Notes

- For the current iteration we decided to ship icons in a bundle (svg + js style), no API or lazy loading is currently happening
- In order to keep the bundle size low we are leveraging the "incon kits" Pro Font Awesome feature, therefore tailoring the icon set that we ship. In order to install the design system users therefore need to configure a `.npmrc` in the application root folder, use the one in this repo as a guideline.
To avoid exposing secrets the Font Awesome Pro token should be configured in a environment variable as `FONTAWESOME_PACKAGE_TOKEN`