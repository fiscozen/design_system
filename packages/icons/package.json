{
  "name": "@fiscozen/icons",
  "version": "0.1.25",
  "description": "Design system icon plugin and component",
  "scripts": {
    "format": "prettier --write src/",
    "test": "exit 0;",
    "build": "exit 0;"
  },
  "main": "src/index.ts",
  "type": "module",
  "keywords": [],
  "author": "Riccardo Agnoletto",
  "license": "ISC",
  "peerDependencies": {
    "vue": "^3.4.13"
  },
  "devDependencies": {
    "@fiscozen/eslint-config": "workspace:^",
    "@fiscozen/prettier-config": "workspace:^",
    "@fiscozen/tsconfig": "workspace:^"
  },
  "dependencies": {
    "@awesome.me/kit-8137893ad3": "^1.0.284",
    "@fortawesome/fontawesome-svg-core": "^6.6.0",
    "@fortawesome/vue-fontawesome": "^3.0.8"
  }
}
