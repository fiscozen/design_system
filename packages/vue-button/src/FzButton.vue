<template>
  <button type="button" :class="classes" @click="onClick" :style="style">{{ label }} </button>
</template>

<script lang="ts" setup>
import { computed } from 'vue';

const props = withDefaults(defineProps<{
  /**
   * The label of the button
   */
  label: string,
  /**
   * primary or secondary button
   */
  primary?: boolean,
  /**
   * size of the button
   */
  size?: 'small' | 'medium' | 'large',
  /**
   * background color of the button
   */
  backgroundColor?: string,

}>(), { primary: false });

const emit = defineEmits<{
  /**
   * simple emit function, always emits 1
   */
  (e: 'click', id: number): void;
}>();

const classes = computed(() => ({
  'storybook-button': true,
  'storybook-button--primary': props.primary,
  'storybook-button--secondary': !props.primary,
  [`storybook-button--${props.size || 'medium'}`]: true,
}));

const style = computed(() => ({
  backgroundColor: props.backgroundColor
}));

const onClick = () => {
  emit("click", 1)
};

</script>

<style>
.storybook-button {
  font-family: 'Nunito Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-weight: 700;
  border: 0;
  border-radius: 3em;
  cursor: pointer;
  display: inline-block;
  line-height: 1;
}
.storybook-button--primary {
  color: white;
  background-color: #1ea7fd;
}
.storybook-button--secondary {
  color: #333;
  background-color: transparent;
  box-shadow: rgba(0, 0, 0, 0.15) 0px 0px 0px 1px inset;
}
.storybook-button--small {
  font-size: 12px;
  padding: 10px 16px;
}
.storybook-button--medium {
  font-size: 14px;
  padding: 11px 20px;
}
.storybook-button--large {
  font-size: 16px;
  padding: 12px 24px;
}

</style>