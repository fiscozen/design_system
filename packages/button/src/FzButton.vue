<template>
  <button type="button" :disabled="disabled" :class="[staticClasses, classes]">
    <div v-if="iconAndLabel" :class="[staticIconClasses, beforeClasses]">
      <slot name="before">
        <FzIcon
          v-if="iconName && iconPosition === 'before'"
          :name="iconName"
          :size="mappedIconSize"
          :variant="iconVariant"
        />
      </slot>
    </div>
    <div :class="containerClass">
      <slot>
        {{ label }}
      </slot>
    </div>
    <div v-if="slots.after || iconAndLabel" :class="[staticIconClasses, afterClasses]">
      <slot name="after">
        <FzIcon
          v-if="iconName && iconPosition === 'after'"
          :name="iconName"
          :size="mappedIconSize"
        />
      </slot>
    </div>
    <span class="hidden h-0 w-0">{{ tooltip }}</span>
  </button>
</template>

<script lang="ts" setup>
import { computed, useSlots } from 'vue'
import { type IconVariant, FzIcon } from '@fiscozen/icons'
import type { ButtonSize, ButtonVariant } from './types'
import { iconSizeMap } from './utils'

const props = withDefaults(
  defineProps<{
    /**
     * The label of the button
     */
    label?: string
    /**
     * The tooltip of the button will be shown on hover
     */
    tooltip?: string
    /**
     * primary or secondary button
     */
    variant?: ButtonVariant
    /**
     * size of the button
     */
    size?: ButtonSize
    /**
     * whether action is enabled or not
     */
    disabled?: boolean
    /**
     * Icon to be displayed. Use fontawesome search here https://fontawesome.com/v6/icons
     * Mind that not all of the icons and variants might be available.
     */
    iconName?: string
    /**
     * Fontawesome icon variant: solid, regular, light, thin. Sharp subvariants are available as well
     */
    iconVariant?: IconVariant
    /**
     * Positioning of the icon
     */
    iconPosition?: 'before' | 'after',
    /**
     * Define the css class for the default slot container
     */
    containerClass?: string
    /**
     * Whether to override internal container classes
     */
    overrideContainerClass?: boolean
  }>(),
  {
    variant: 'primary',
    size: 'md',
    disabled: false,
    iconPosition: 'before',
  }
)
console.log('FzButton', props)
const slots = useSlots()

const customVariantClasses = computed(() => {
  switch (props.variant) {
    case 'primary':
      return {
        'bg-blue-500': true,
        'hover:bg-blue-600': true,
        'disabled:bg-blue-200': true,
        'text-core-white': true,
        'focus:bg-blue-500': !props.disabled
      }
      break
    case 'secondary':
      return {
        'text-grey-500': true,
        'bg-core-white': true,
        '!border-grey-200': true,
        'hover:bg-grey-100': !props.disabled,
        'focus:!border-blue-600': !props.disabled,
        'disabled:text-grey-100': true,
      }
      break
    case 'invisible':
      return {
        'text-grey-500': true,
        'bg-transparent': true,
        'border-transparent': true,
        'hover:bg-grey-100': !props.disabled,
        'focus:!border-blue-600': !props.disabled,
        'disabled:text-grey-100': true
      }
      break
    case 'danger':
      return {
        'text-core-white': true,
        'temporary-bg-red': true,
        'temporary-border-red': !props.disabled,
        'bg-semantic-error': true
      }
      break
    case 'success':
      return {
        'text-core-white': true,
        'temporary-bg-green': true,
        'temporary-border-green': !props.disabled,
        'bg-semantic-success': true
      }
      break
    default:
      return {}
      break
  }
})

const iconAndLabel = computed(() => Boolean((props.label || slots.default) && props.iconName))
const mappedIconSize = computed(() => iconSizeMap[props.size])

const staticClasses = [
  'relative',
  'rounded',
  'flex',
  'flex-shrink',
  'items-center',
  'justify-center',
  'font-medium',
  'border-1',
  'border-transparent'
]

const staticIconClasses = ['flex', 'items-center', 'justify-items-center']

const classes = computed(() => ({
  'h-24 text-xs': props.size === 'xs',
  'h-28 text-sm': props.size === 'sm',
  'h-32': props.size === 'md',
  'h-40 text-lg': props.size === 'lg',
  'focus:border-blue-600': !props.disabled,
  'focus:border-solid': !props.disabled,
  'focus:border-1': !props.disabled,
  'px-12': props.size === 'xs' && !iconAndLabel.value,
  'px-14': props.size === 'sm' && !iconAndLabel.value,
  'px-16': props.size === 'md' && !iconAndLabel.value,
  'px-20': props.size === 'lg' && !iconAndLabel.value,
  ...customVariantClasses.value
}))

const beforeClasses = computed(() => ({
  'mr-8': props.size === 'lg',
  'mr-6': props.size === 'md',
  'mr-4': props.size === 'sm' || props.size === 'xs',
  'pl-12': props.size === 'lg' && iconAndLabel.value,
  'pl-10': (props.size === 'md' || props.size === 'sm') && iconAndLabel.value,
  'pl-8': props.size === 'xs' && iconAndLabel.value
}))

const afterClasses = computed(() => ({
  'ml-8': props.size === 'lg',
  'ml-6': props.size === 'md',
  'ml-4': props.size === 'sm' || props.size === 'xs',
  'pr-12': props.size === 'lg' && iconAndLabel.value,
  'pr-10': (props.size === 'md' || props.size === 'sm') && iconAndLabel.value,
  'pr-8': props.size === 'xs' && iconAndLabel.value
}))

const containerClass = computed(() => {
  if (props.overrideContainerClass) {
    return props.containerClass
  }

  return ["truncate", props.containerClass]
});
</script>

<style>
.temporary-bg-red:hover {
  background-color: #aa2f2f;
}
.temporary-bg-red:disabled {
  background-color: #f8baba;
}
.temporary-border-red:focus {
  border-color: #aa2f2f;
}
.temporary-bg-green:hover {
  background-color: #0b7763;
}
.temporary-bg-green:disabled {
  background-color: #b5d8ce;
}
.temporary-border-green:focus {
  border-color: #0b7763;
}
</style>
