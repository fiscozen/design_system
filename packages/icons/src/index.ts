import { library } from '@fortawesome/fontawesome-svg-core'
import { all } from '@awesome.me/kit-8137893ad3/icons'
import FzIcon from './FzIcon.vue'

const IconPlugin = {
  install(app) {
    library.add(...all)
    app.component('fz-icon', FzIcon)
  }
}

export { FzIcon, IconPlugin }

export type { IconVariant, IconSize } from './types'
