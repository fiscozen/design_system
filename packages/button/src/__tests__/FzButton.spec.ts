import { describe, it, expect, vi } from 'vitest'

import { mount } from '@vue/test-utils'
import FzButton from '../FzButton.vue'
import { FzIcon } from '@fiscozen/icons'

describe.concurrent('FzButton', () => {
  describe('Basic Rendering', () => {
    it('should render with label', () => {
      const wrapper = mount(FzButton, {
        props: {
          label: 'Test Button',
        }
      })

      expect(wrapper.text()).toContain('Test Button')
      expect(wrapper.find('button').exists()).toBe(true)
    })

    it('should render with default slot', () => {
      const wrapper = mount(FzButton, {
        slots: {
          default: 'Slot Content'
        }
      })

      expect(wrapper.text()).toContain('Slot Content')
    })

    it('should prioritize slot over label', () => {
      const wrapper = mount(FzButton, {
        props: {
          label: 'Label Text'
        },
        slots: {
          default: 'Slot Text'
        }
      })

      expect(wrapper.text()).toContain('Slot Text')
      expect(wrapper.text()).not.toContain('Label Text')
    })
  })

  describe('Variants', () => {
    it('should render primary variant with correct classes', () => {
      const wrapper = mount(FzButton, {
        props: {
          variant: 'primary',
          label: 'Primary'
        }
      })

      const button = wrapper.find('button')
      expect(button.classes()).toContain('bg-blue-500')
      expect(button.classes()).toContain('text-core-white')
    })

    it('should render secondary variant with correct classes', () => {
      const wrapper = mount(FzButton, {
        props: {
          variant: 'secondary',
          label: 'Secondary'
        }
      })

      const button = wrapper.find('button')
      expect(button.classes()).toContain('bg-core-white')
      expect(button.classes()).toContain('text-grey-500')
    })

    it('should render invisible variant with correct classes', () => {
      const wrapper = mount(FzButton, {
        props: {
          variant: 'invisible',
          label: 'Invisible'
        }
      })

      const button = wrapper.find('button')
      expect(button.classes()).toContain('bg-transparent')
      expect(button.classes()).toContain('border-transparent')
    })

    it('should render danger variant with correct classes', () => {
      const wrapper = mount(FzButton, {
        props: {
          variant: 'danger',
          label: 'Danger'
        }
      })

      const button = wrapper.find('button')
      expect(button.classes()).toContain('bg-semantic-error-200')
      expect(button.classes()).toContain('text-core-white')
    })

    it('should render success variant with correct classes', () => {
      const wrapper = mount(FzButton, {
        props: {
          variant: 'success',
          label: 'Success'
        }
      })

      const button = wrapper.find('button')
      expect(button.classes()).toContain('bg-semantic-success-200')
      expect(button.classes()).toContain('text-core-white')
    })
  })

  describe('Environment', () => {
    it('should render backoffice environment with correct height', () => {
      const wrapper = mount(FzButton, {
        props: {
          environment: 'backoffice',
          label: 'Backoffice Button'
        }
      })

      const button = wrapper.find('button')
      expect(button.classes()).toContain('h-32')
      expect(button.classes()).not.toContain('text-lg')
    })

    it('should render frontoffice environment with correct height', () => {
      const wrapper = mount(FzButton, {
        props: {
          environment: 'frontoffice',
          label: 'Frontoffice Button'
        }
      })

      const button = wrapper.find('button')
      expect(button.classes()).toContain('h-44')
      expect(button.classes()).not.toContain('text-lg')
    })

    it('should use frontoffice as default environment', () => {
      const wrapper = mount(FzButton, {
        props: {
          label: 'Default Button'
        }
      })

      const button = wrapper.find('button')
      expect(button.classes()).toContain('h-44')
    })
  })

  describe('Size Deprecation', () => {
    it('should log deprecation warning for any size prop', async () => {
      const warnSpy = vi.spyOn(console, 'warn').mockImplementation(() => {})
      
      const wrapper = mount(FzButton, {
        props: {
          size: 'lg',
          label: 'Legacy Button'
        }
      })

      await wrapper.vm.$nextTick()

      expect(warnSpy).toHaveBeenCalledWith(
        expect.stringContaining('[FzButton] The "size" prop is deprecated')
      )
      
      warnSpy.mockRestore()
    })

    it('should log deprecation warning for md size', async () => {
      const warnSpy = vi.spyOn(console, 'warn').mockImplementation(() => {})
      
      const wrapper = mount(FzButton, {
        props: {
          size: 'md',
          label: 'MD Button'
        }
      })

      await wrapper.vm.$nextTick()

      expect(warnSpy).toHaveBeenCalledWith(
        expect.stringContaining('[FzButton] The "size" prop is deprecated')
      )
      
      warnSpy.mockRestore()
    })

    it('should map lg size to frontoffice environment', () => {
      const wrapper = mount(FzButton, {
        props: {
          size: 'lg',
          label: 'LG Button'
        }
      })

      const button = wrapper.find('button')
      expect(button.classes()).toContain('h-44')
      expect(button.classes()).not.toContain('text-lg')
    })

    it('should map md size to backoffice environment', () => {
      const wrapper = mount(FzButton, {
        props: {
          size: 'md',
          label: 'MD Button'
        }
      })

      const button = wrapper.find('button')
      expect(button.classes()).toContain('h-32')
      expect(button.classes()).not.toContain('text-lg')
    })
  })

  describe('Icons', () => {
    it('should render icon before label', () => {
      const wrapper = mount(FzButton, {
        props: {
          label: 'With Icon',
          iconName: 'bell',
          iconPosition: 'before'
        }
      })

      const icon = wrapper.findComponent(FzIcon)
      expect(icon.exists()).toBe(true)
      const button = wrapper.find('button')
      expect(button.classes()).toContain('gap-8')
      expect(button.classes()).toContain('px-12')
    })

    it('should render icon after label', () => {
      const wrapper = mount(FzButton, {
        props: {
          label: 'With Icon',
          iconName: 'bell',
          iconPosition: 'after'
        }
      })

      const icon = wrapper.findComponent(FzIcon)
      expect(icon.exists()).toBe(true)
      const button = wrapper.find('button')
      expect(button.classes()).toContain('gap-8')
      expect(button.classes()).toContain('px-12')
    })

    it('should not render icon without iconName', () => {
      const wrapper = mount(FzButton, {
        props: {
          label: 'No Icon'
        }
      })

      const icon = wrapper.findComponent(FzIcon)
      expect(icon.exists()).toBe(false)
      const button = wrapper.find('button')
      expect(button.classes()).toContain('gap-8')
      expect(button.classes()).toContain('px-12')
    })

    it('should use before slots when provided', () => {
      const wrapper = mount(FzButton, {
        props: {
          label: 'Button',
          iconName: 'bell'
        },
        slots: {
          before: '<span class="custom-before">Custom Before</span>'
        }
      })

      expect(wrapper.find('.custom-before').exists()).toBe(true)
    })

    it('should use after slots when provided', () => {
      const wrapper = mount(FzButton, {
        props: {
          label: 'Button',
          iconName: 'bell'
        },
        slots: {
          after: '<span class="custom-after">Custom After</span>'
        }
      })

      expect(wrapper.find('.custom-after').exists()).toBe(true)
    })
  })

  describe('Disabled State', () => {
    it('should apply disabled attribute when disabled', () => {
      const wrapper = mount(FzButton, {
        props: {
          label: 'Disabled',
          disabled: true
        }
      })

      expect(wrapper.find('button').attributes('disabled')).toBeDefined()
    })

    it('should apply disabled classes for primary variant', () => {
      const wrapper = mount(FzButton, {
        props: {
          variant: 'primary',
          label: 'Disabled Primary',
          disabled: true
        }
      })

      const button = wrapper.find('button')
      expect(button.classes()).toContain('disabled:bg-blue-200')
    })

    it('should not apply hover classes when disabled', () => {
      const wrapper = mount(FzButton, {
        props: {
          variant: 'primary',
          label: 'Disabled',
          disabled: true
        }
      })

      const button = wrapper.find('button')
      expect(button.classes()).not.toContain('focus:bg-blue-500')
    })

    it('should not emit click when disabled', () => {
      const wrapper = mount(FzButton, {
        props: {
          label: 'Disabled',
          disabled: true
        }
      })
      wrapper.trigger('click')

      const clickEvs = wrapper.emitted('click')
      expect(clickEvs).toBeUndefined()
    })
  })

  describe('Events', () => {
    it('should emit native click event', () => {
      const wrapper = mount(FzButton, {
        props: {
          label: 'Clickable'
        }
      })
      wrapper.trigger('click')

      const clickEvs = wrapper.emitted('click')
      expect(clickEvs).toHaveLength(1)
    })
  })

  describe('CSS Classes', () => {
    it('should apply static classes', () => {
      const wrapper = mount(FzButton, {
        props: {
          label: 'Button'
        }
      })

      const button = wrapper.find('button')
      expect(button.classes()).toContain('relative')
      expect(button.classes()).toContain('rounded')
      expect(button.classes()).toContain('flex')
      expect(button.classes()).toContain('items-center')
      expect(button.classes()).toContain('justify-center')
      expect(button.classes()).toContain('font-normal')
      expect(button.classes()).toContain('!text-[16px]')
      expect(button.classes()).toContain('!leading-[20px]')
      expect(button.classes()).toContain('border-1')
    })

    it('should apply focus classes when not disabled', () => {
      const wrapper = mount(FzButton, {
        props: {
          label: 'Button',
          disabled: false
        }
      })

      const button = wrapper.find('button')
      expect(button.classes()).toContain('focus:!border-blue-600')
    })

    it('should apply containerClass when provided', () => {
      const wrapper = mount(FzButton, {
        props: {
          label: 'Button',
          containerClass: 'custom-class'
        }
      })

      const container = wrapper.find('div.truncate')
      expect(container.classes()).toContain('custom-class')
    })

    it('should override container classes when overrideContainerClass is true', () => {
      const wrapper = mount(FzButton, {
        props: {
          label: 'Button',
          containerClass: 'only-custom',
          overrideContainerClass: true
        }
      })

      // When overrideContainerClass is true, only containerClass is applied
      const html = wrapper.html()
      expect(html).toContain('only-custom')
      expect(html).not.toContain('truncate')
    })
  })

  describe('Combinations', () => {
    it('should handle danger variant with frontoffice environment', () => {
      const wrapper = mount(FzButton, {
        props: {
          variant: 'danger',
          environment: 'frontoffice',
          label: 'Large Danger'
        }
      })

      const button = wrapper.find('button')
      expect(button.classes()).toContain('bg-semantic-error-200')
      expect(button.classes()).toContain('h-44')
      expect(button.classes()).not.toContain('text-lg')
    })

    it('should handle success variant with icon and backoffice environment', () => {
      const wrapper = mount(FzButton, {
        props: {
          variant: 'success',
          environment: 'backoffice',
          label: 'Success',
          iconName: 'check',
          iconPosition: 'before'
        }
      })

      const button = wrapper.find('button')
      expect(button.classes()).toContain('bg-semantic-success-200')
      expect(button.classes()).toContain('h-32')
    })

    it('should handle secondary variant disabled with icon', () => {
      const wrapper = mount(FzButton, {
        props: {
          variant: 'secondary',
          disabled: true,
          label: 'Disabled Secondary',
          iconName: 'bell'
        }
      })

      const button = wrapper.find('button')
      expect(button.classes()).toContain('bg-core-white')
      expect(button.classes()).toContain('disabled:text-grey-200')
      expect(button.attributes('disabled')).toBeDefined()
    })
  })

  describe('Snapshots', () => {
    it('should match snapshot - basic button', () => {
      const wrapper = mount(FzButton, {
        props: {
          label: 'some text',
          disabled: false,
        }
      })

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('should match snapshot - with icon', () => {
      const wrapper = mount(FzButton, {
        props: {
          tooltip: 'some text',
          disabled: false,
          iconName: 'bell',
          label: 'some label',
        }
      })

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('should match snapshot - danger variant', () => {
      const wrapper = mount(FzButton, {
        props: {
          variant: 'danger',
          label: 'Danger Button'
        }
      })

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('should match snapshot - success variant', () => {
      const wrapper = mount(FzButton, {
        props: {
          variant: 'success',
          label: 'Success Button'
        }
      })

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('should match snapshot - backoffice environment', () => {
      const wrapper = mount(FzButton, {
        props: {
          environment: 'backoffice',
          label: 'Backoffice Button'
        }
      })

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('should match snapshot - frontoffice environment with icon', () => {
      const wrapper = mount(FzButton, {
        props: {
          environment: 'frontoffice',
          label: 'Frontoffice Button',
          iconName: 'star',
          iconPosition: 'after'
        }
      })

      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
