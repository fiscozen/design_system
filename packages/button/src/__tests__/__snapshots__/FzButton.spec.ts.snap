// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FzButton > Snapshots > should match snapshot - backoffice environment 1`] = `
"<button type="button" class="relative rounded flex flex-shrink items-center justify-center font-normal border-1 border-transparent gap-8 h-44 px-12 min-w-96 bg-blue-500 text-core-white hover:bg-blue-600 focus:bg-blue-500 focus:!border-blue-600 disabled:bg-blue-200">
  <!--v-if-->
  <div class="truncate">some text</div>
  <!--v-if-->
</button>"
`;

exports[`FzButton > Snapshots > should match snapshot - backoffice environment 2`] = `
"<button type="button" class="relative rounded flex flex-shrink items-center justify-center font-normal border-1 border-transparent gap-8 h-44 px-12 min-w-96 bg-blue-500 text-core-white hover:bg-blue-600 focus:bg-blue-500 focus:!border-blue-600 disabled:bg-blue-200">
  <div class="flex items-center justify-items-center">
    <div class="flex items-center justify-center w-[20px] h-[20px]"><svg class="svg-inline--fa fa-bell h-[16px]" aria-hidden="true" focusable="false" data-prefix="far" data-icon="bell" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512">
        <path class="" fill="currentColor" d="M224 0c-17.7 0-32 14.3-32 32l0 19.2C119 66 64 130.6 64 208l0 25.4c0 45.4-15.5 89.5-43.8 124.9L5.3 377c-5.8 7.2-6.9 17.1-2.9 25.4S14.8 416 24 416l400 0c9.2 0 17.6-5.3 21.6-13.6s2.9-18.2-2.9-25.4l-14.9-18.6C399.5 322.9 384 278.8 384 233.4l0-25.4c0-77.4-55-142-128-156.8L256 32c0-17.7-14.3-32-32-32zm0 96c61.9 0 112 50.1 112 112l0 25.4c0 47.9 13.9 94.6 39.7 134.6L72.3 368C98.1 328 112 281.3 112 233.4l0-25.4c0-61.9 50.1-112 112-112zm64 352l-64 0-64 0c0 17 6.7 33.3 18.7 45.3s28.3 18.7 45.3 18.7s33.3-6.7 45.3-18.7s18.7-28.3 18.7-45.3z"></path>
      </svg></div>
  </div>
  <div class="truncate">some label</div>
  <!--v-if-->
</button>"
`;

exports[`FzButton > Snapshots > should match snapshot - backoffice environment 3`] = `
"<button type="button" class="relative rounded flex flex-shrink items-center justify-center font-normal border-1 border-transparent gap-8 h-44 px-12 min-w-96 bg-semantic-error-200 text-core-white hover:bg-semantic-error-300 focus:bg-semantic-error-200 focus:!border-semantic-error-300 disabled:bg-semantic-error-100">
  <!--v-if-->
  <div class="truncate">Danger Button</div>
  <!--v-if-->
</button>"
`;

exports[`FzButton > Snapshots > should match snapshot - backoffice environment 4`] = `
"<button type="button" class="relative rounded flex flex-shrink items-center justify-center font-normal border-1 border-transparent gap-8 h-44 px-12 min-w-96 bg-semantic-success-200 text-core-white hover:bg-semantic-success-300 focus:bg-semantic-success-200 focus:!border-semantic-success-300 disabled:bg-semantic-success-100">
  <!--v-if-->
  <div class="truncate">Success Button</div>
  <!--v-if-->
</button>"
`;

exports[`FzButton > Snapshots > should match snapshot - backoffice environment 5`] = `
"<button type="button" class="relative rounded flex flex-shrink items-center justify-center font-normal border-1 border-transparent gap-8 h-32 px-12 min-w-96 bg-blue-500 text-core-white hover:bg-blue-600 focus:bg-blue-500 focus:!border-blue-600 disabled:bg-blue-200">
  <!--v-if-->
  <div class="truncate">Backoffice Button</div>
  <!--v-if-->
</button>"
`;

exports[`FzButton > Snapshots > should match snapshot - frontoffice environment with icon 1`] = `
"<button type="button" class="relative rounded flex flex-shrink items-center justify-center font-normal border-1 border-transparent gap-8 h-44 px-12 min-w-96 bg-blue-500 text-core-white hover:bg-blue-600 focus:bg-blue-500 focus:!border-blue-600 disabled:bg-blue-200">
  <!--v-if-->
  <div class="truncate">Frontoffice Button</div>
  <div class="flex items-center justify-items-center">
    <div class="flex items-center justify-center w-[20px] h-[20px]"><svg class="svg-inline--fa fa-star h-[16px]" aria-hidden="true" focusable="false" data-prefix="far" data-icon="star" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512">
        <path class="" fill="currentColor" d="M287.9 0c9.2 0 17.6 5.2 21.6 13.5l68.6 141.3 153.2 22.6c9 1.3 16.5 7.6 19.3 16.3s.5 18.1-5.9 24.5L433.6 328.4l26.2 155.6c1.5 9-2.2 18.1-9.7 23.5s-17.3 6-25.3 1.7l-137-73.2L151 509.1c-8.1 4.3-17.9 3.7-25.3-1.7s-11.2-14.5-9.7-23.5l26.2-155.6L31.1 218.2c-6.5-6.4-8.7-15.9-5.9-24.5s10.3-14.9 19.3-16.3l153.2-22.6L266.3 13.5C270.4 5.2 278.7 0 287.9 0zm0 79L235.4 187.2c-3.5 7.1-10.2 12.1-18.1 13.3L99 217.9 184.9 303c5.5 5.5 8.1 13.3 6.8 21L171.4 443.7l105.2-56.2c7.1-3.8 15.6-3.8 22.6 0l105.2 56.2L384.2 324.1c-1.3-7.7 1.2-15.5 6.8-21l85.9-85.1L358.6 200.5c-7.8-1.2-14.6-6.1-18.1-13.3L287.9 79z"></path>
      </svg></div>
  </div>
</button>"
`;
